### PR TITLE
lapack: Update to 3.12.0

### PR DESCRIPTION
**Pull request by Dave-Allured**
* Update lapack 3.11.0 --> 3.12.0.
* Remove broken subport lapack-manpages.
* Add note redirecting to on-line docs.
* Lint fix.  Gather portgroup lines at top of file.
* Update to portgroup cmake 1.1.
* Add myself as co-maintainer.

Build system for man pages changed.  I could not figure out how to fix. Previous subport used unconventional method, skipped configure, used doxygen directly. Someone who understands Macports better than me, please fix that subport. See github history for 2024 January 20.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.7.1 21G920 x86_64
Xcode 14.2 / Command Line Tools 14.2.0

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tried a full install with `sudo port -vs install`?
- [ ] tested basic functionality of all binary files? **(Library only, no executables.)**
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by pmetzger** [comment]
LGTM. We'll see what @tenomoto thinks.

**Comment by Dave-Allured** [comment]
> LGTM. We'll see what @tenomoto thinks.

Thanks, Perry.  To be clear, my intention is to get the library updated now.  The man page fix wait for a separate PR.

File: math/lapack/Portfile [hunk 1]
```diff
@@ -2,21 +2,28 @@
 
 PortSystem          1.0
 PortGroup           github 1.0
+PortGroup           compilers 1.0
+PortGroup           cmake 1.0
```
Comment (reneeotten) [inline math/lapack/Portfile:6]: this should use the latest version of the `cmake` PortGroup, which is "1.1"

File: math/lapack/Portfile [hunk 2]
```diff
@@ -2,21 +2,28 @@
...
-github.setup        Reference-LAPACK lapack 3.11.0 v
+github.setup        Reference-LAPACK lapack 3.12.0 v
 if {${subport} eq ${name}} {
-    PortGroup           compilers 1.0
     compilers.choose    cc fc f77 f90
     compilers.setup     require_fortran -clang
     compilers.add_gcc_rpath_support no
 
-    PortGroup           cmake 1.0
     cmake.out_of_source yes
```
Comment (reneeotten) [inline math/lapack/Portfile:14]: this can/should be removed as it is the default in the most recent `cmake` PortGroup version

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: math/lapack/Portfile (hunk 1 above)
Comment (Dave-Allured) [reply math/lapack/Portfile]: Done.

File: math/lapack/Portfile (hunk 2 above)
Comment (Dave-Allured) [reply math/lapack/Portfile]: Done.

**Comment by pmetzger** [comment]
Hi @Dave-Allured! One last thing, can you please rebase your commits so we end up with a single clean commit?

**Comment by Dave-Allured** [comment]
@pmetzger, will do.  It might take a while because I am just getting started using git.

**Comment by ryandesign** [comment]
> can you please rebase your commits so we end up with a single clean commit?

It's not necessary to wait for the submitter to do that. You can do it here in the GitHub web interface by pressing the "Squash and merge" button instead of "Rebase and merge". Just be sure to edit the commit message to remove extraneous or duplicate lines.

**Comment by pmetzger** [comment]
@ryandesign Yes, and we sometimes do that, but when you're merging a lot of PRs per day, it is helpful when the submitter is doing the bulk of the work.

**Pull request merged**